### PR TITLE
feat(api): allow ${env} reference in custom-roles.json

### PR DESCRIPTION
**Pull request by joekiller**
#### Description of changes

Allow using `${env}` in custom-roles.json to reference the current Amplify CLI environment.

Also lint --fix was applied.

#### Issue #, if available

Documentation update: https://github.com/aws-amplify/docs/pull/4591

#### Description of how you validated changes

Added tests.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [x] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by joekiller** [comment]
@marcvberg I have removed the yarn.lock changes

File: packages/amplify-e2e-tests/src/__tests__/api_10.test.ts [hunk 1]
```diff
@@ -125,10 +127,10 @@ describe('amplify add api (GraphQL)', () => {
     expect(beforeAdminConfig).toMatchSnapshot();
 
     const customRolesConfig = {
-      adminRoleNames: ['myAdminRoleName'],
+      adminRoleNames: ['myAdminRoleName', 'myEnvFunction-${env}'],
     };
     setCustomRolesConfig(projRoot, name, customRolesConfig);
-    await apiGqlCompile(projRoot);
+    await apiGqlCompile(projRoot, true);
```
Comment (marcvberg) [inline packages/amplify-e2e-tests/src/__tests__/api_10.test.ts:133]: Why is it necessary to use true here instead of the default false?

Comment (joekiller) [reply packages/amplify-e2e-tests/src/__tests__/api_10.test.ts:133]: The graphql compiler seems to use what is installed in node_modules unless true is set and then it uses the compiler which includes the code I updated.

Comment (joekiller) [reply packages/amplify-e2e-tests/src/__tests__/api_10.test.ts:133]: The compiler is resolved here: https://github.com/aws-amplify/amplify-category-api/blob/c79232ff8a2f7809447f51233a2dbe0046c25e47/packages/amplify-e2e-core/src/categories/api.ts#L25

And it uses either the internal or pre installed CLI determined off the boolean flag. 

https://github.com/aws-amplify/amplify-category-api/blob/c79232ff8a2f7809447f51233a2dbe0046c25e47/packages/amplify-e2e-core/src/index.ts#L30-L35

**Comment by joekiller** [comment]
@marcvberg anything else I can do here? this feature would be useful for my team already :)

**Comment by joekiller** [comment]
@marcvberg, @sundersc, or @alharris-at seems the docs have shipped but this has yet to merge.

**Comment by alharris-at** [comment]
Hi @joekiller, just committed an additional merge conflict change, and reviewing this PR now. Also kicking off e2e tests, if those all look good, we'll get this merged in.

Thank you for following up.

**Review by alharris-at — APPROVED** [review]

**Comment by alharris-at** [comment]
E2E run: https://app.circleci.com/pipelines/github/aws-amplify/amplify-category-api?branch=run-e2e%2Fjoekiller%2Fallow-env-reference-in-custom-role-file

**Review by alharris-at — APPROVED** [review]

**Review by AaronZyLee — APPROVED** [review]

**Pull request merged**